src/utils/Logging.ts

```diff
@@ -703,67 +704,70 @@ export default class Logging {
-        // Anonymize
-        message.replace(new RegExp(sensitiveData, 'gi'), Constants.ANONYMIZED_VALUE);
+        if (new RegExp(sensitiveData + '[:=](.*)', 'gi').test(message)) { // If it contains sensitive data and it matches the RegExp, Anonymize
+          message = message.substring(0,sensitiveData.length + 1) + Constants.ANONYMIZED_VALUE;
+          return message;
+        }
       }
-    } else if (Array.isArray(message)) {
+      return message;
+    } else if (Array.isArray(message)) { // If the message is an array, apply the anonymizeSensitiveData function for each item
+      const anonymizedMessage = [];
       for (const item of message) {
-        Logging.anonymizeSensitiveData(item);
+        anonymizedMessage.push(Logging.anonymizeSensitiveData(item));
       }
-    } else if (typeof message === 'object') {
+      return anonymizedMessage;
+    } else if (typeof message === 'object') { // If the message is an object
       for (const key of Object.keys(message)) {
-        // String?
-        if (typeof message[key] === 'string') {
-          for (const sensitiveData of Constants.SENSITIVE_DATA) {
-            if (key.toLocaleLowerCase() === sensitiveData.toLocaleLowerCase()) {
-              // Anonymize
-              message[key] = Constants.ANONYMIZED_VALUE;
-            }
-          }
-          // Check query string
-          const dataParts: string[] = message[key].split('&');
-          if (dataParts.length > 1) {
-            for (let i = 0; i < dataParts.length; i++) {
-              const dataPart = dataParts[i];
-              for (const sensitiveData of Constants.SENSITIVE_DATA) {
-                if (dataPart.toLowerCase().startsWith(sensitiveData.toLocaleLowerCase())) {
-                  // Anonymize
-                  dataParts[i] = dataPart.substring(0, sensitiveData.length + 1) + Constants.ANONYMIZED_VALUE;
-                }
-              }
-            }
-            message[key] = dataParts.join('&');
-          }
-        } else {
-          Logging.anonymizeSensitiveData(message[key]);
+        if (typeof message[key] === 'string' && Constants.SENSITIVE_DATA.filter((sensitiveData) => key.toLocaleLowerCase() === sensitiveData.toLocaleLowerCase()).length > 0) {
+          // If the key indicates sensitive data and the value is a string, Anonymize the value
+          message[key] = Constants.ANONYMIZED_VALUE;
+        } else { // Otherwise, apply the anonymizeSensitiveData function
+          message[key] = Logging.anonymizeSensitiveData(message[key]);
         }
       }
-    } else {
-      // Log
-      Logging.logError({
-        tenantID: Constants.DEFAULT_TENANT,
-        type: LogType.SECURITY,
-        module: MODULE_NAME,
-        method: 'anonymizeSensitiveData',
-        action: ServerAction.LOGGING,
-        message: 'No matching object type for log message anonymisation',
-        detailedMessages: { message: message }
-      });
+      return message;
```

Comment:
Why do you need to return the message? The object instance is cloned and the recursion do change directly the instance.

Reply:
I modified the function to use returns everywhere because I noticed that it was not doing anything when the message was of type string, meaning that -  the string that was being saved in the Log object was always the original string without any modification (data was not anonymized).
I believe this has to do with the fact that in JS/TS strings are passed by value, not by reference, so the reference to the string in the original object was not updated to the modified value.

Reply:
> I modified the function to use returns everywhere because I noticed that it was not doing anything when the message was of type string, meaning that - the string that was being saved in the Log object was always the original string without any modification (data was not anonymized).
> I believe this has to do with the fact that in JS/TS strings are passed by value, not by reference, so the reference to the string in the original object was not updated to the modified value.

Fair enough then.